### ISSUE TITLE
[core] Variable type is inferred incorrectly

### ISSUE THREAD

**Issue by bagipro** [skylot/jadx#1197]
Hey

Class `androidx.documentfile.provider.DocumentsContractApi19`
```java
    private static String queryForString(Context context, Uri uri, String str, @Nullable String str2) {
        AutoCloseable autoCloseable = null; // `Cursor` should be here
        try {
            autoCloseable = context.getContentResolver().query(uri, new java.lang.String[]{str}, null, null, null);
            if (autoCloseable.moveToFirst() && !autoCloseable.isNull(0)) {
                return autoCloseable.getString(0); // `AutoCloseable` doesn't have `getString` method
            }
```

APK https://drive.google.com/file/d/1qC3tlWs9AtPBpyS6iU9kcSphLryfJOxi/view

**Comment by skylot** [comment]
This is a known issue in try-catch processing, see https://github.com/skylot/jadx/issues/920#issuecomment-718842646 for details.
Right now I am working on rewrite of try-catch handling and this case is already resolved, but complete rewrite not finished yet, and some tests still failing, so you need to wait, sorry.

**Comment by skylot** [comment]
Fixed in 7c0671c81b1512bf5b56fc4962f89be72d413f27.
